### PR TITLE
Unit test added for Validation Functions

### PR DESCRIPTION
**Pull request by skwong25**
Adds simple unit tests for validation functions, last extracted from the router files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "scripts": {
     "start": "node src/buildServer.js",
-    "test": "mocha 'test/apiTest.js' --exit"
+    "test": "mocha 'test/catRoutes.test.js' --exit",
```
Comment (fractalwrench) [inline package.json:8]:
```suggestion
    "test": "mocha 'test/catRoutes.test.js' --exit && jest"
```

I'd suggest using [bash's AND operator](https://stackoverflow.com/questions/4510640/what-is-the-purpose-of-in-a-shell-command) (&&) to run the jest tests after mocha. This has the added bonus of running them on CI and makes it harder to forget to run them.

File: test/validationFunctions.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const generateErr404 = validate.generateErr404; 
+const checkObjFormat = validate.checkObjFormat; 
+const checkObjKeys = validate.checkObjKeys; 
+const checkBreedObjKeys = validate.checkBreedObjKeys; 
+const checkObjValues = validate.checkObjValues; 
+ 
+
+describe( "checks for invalid data type", () => {
+    
+    let [ notNumber, number ]= [ "foo", 10 ];
```
Comment (fractalwrench) [inline test/validationFunctions.test.js:17]: style nit: the extra whitespace here isn't part of standard JS style

File: test/catRoutes.test.js [hunk 3]
```diff
@@ -199,31 +199,3 @@ describe('POST /cats', function () {
             .expect(400,'Error: Request data is not a valid object. ', done)
     }) 
 })
-
-describe('GET /breeds', function () {
```
Comment (fractalwrench) [inline test/catRoutes.test.js:203]: why do these tests need to be removed? It'd be preferable to keep them in unless there's a good reason

File: test/validationFunctions.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    let [ notString, string ] = [ 10 , "foo" ];
+
+    test('returns true for an invalid string', () => { 
+        expect(isInvalidString(notString)).toBe(true); 
+        expect(isInvalidString(string)).toBe(false); 
+    });
+
+    test('returns true for an invalid number', () => {
```
Comment (fractalwrench) [inline test/validationFunctions.test.js:25]: null/undefined would be good ones to check too :)

File: test/validationFunctions.test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        expect(isInvalidNum(notNumber)).toBeTruthy;
+        expect(isInvalidNum(number)).toBeFalsey;
+    });
+});
+
+describe( "error generator functions should return errors", () => {
```
Comment (fractalwrench) [inline test/validationFunctions.test.js:31]: nice 👍 this is a comprehensive test

**Review by fractalwrench — COMMENTED** [review]
I encountered the following message when running the tests locally:

>A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --runInBand --detectOpenHandles to find leaks.

I also noticed that `test/breedRoutes.test.js` is being included in the Jest tests. Perhaps Jest just grabs everything from the `test` folder by default and attempts to run it, which causes failures, as these ones expect to be run via mocha? You could confirm that by looking at the Jest documentation and seeing if there's a way to customise what test files are used for input.

**Comment by skwong25** [comment]
Yes that is the error message I have been grappling with all day and gave up in the end. :D

File: test/catRoutes.test.js (hunk 3 above)
Comment (skwong25) [reply test/catRoutes.test.js:203]: Tests for the breed router routes are now in their own folder.

**Pull request merged**